lib/prom_ex/ets_cron_flusher.ex

```diff
@@ -0,0 +1,68 @@
+defmodule PromEx.ETSCronFlusher do
+  @moduledoc """
+  This module is used to regularly flush ETS of any buffered distribution
+  type metrics (see https://github.com/beam-telemetry/telemetry_metrics_prometheus_core/blob/main/lib/core.ex#L25-L28)
+  for more information. At the moment the flush interval is not configurable
+  but that could change in the future.
+  """
+
+  use GenServer
+
+  @doc """
+  Used to start the `PromEx.ETSCronFlusher` process.
+  """
+  @spec start_link(opts :: keyword()) :: GenServer.on_start()
+  def start_link(opts) do
+    {name, remaining_opts} = Keyword.pop(opts, :name)
+
+    state =
+      remaining_opts
+      |> Map.new()
+      |> Map.put(:timer_ref, nil)
+
+    GenServer.start_link(__MODULE__, state, name: name)
+  end
+
+  @doc """
+  This function will cancel the existing cron timer and schedule a new
+  new. The reason for this being that if metrics scrapes occur regularly,
+  there is no need to flush via the cron job. Cron ETS flushes should
+  only occur if the metrics endpoint for whatever reason fails to be
+  scraped so that ETS is not leaking memeory.
+  """
+  def defer_ets_flush(instance) do
+    GenServer.cast(instance, :defer_ets_flush)
+  end
+
+  @impl true
+  def init(state) do
+    {:ok, state, {:continue, :start_cron_flusher}}
+  end
+
+  @impl true
+  def handle_continue(:start_cron_flusher, state) do
+    timer_ref = schedule_flush()
+    {:noreply, %{state | timer_ref: timer_ref}}
+  end
+
+  @impl true
+  def handle_info(:flush_ets, state) do
+    PromEx.get_metrics(state.prom_ex_module)
+
+    timer_ref = schedule_flush()
+    {:noreply, %{state | timer_ref: timer_ref}}
+  end
+
+  @impl true
+  def handle_cast(:defer_ets_flush, state) do
+    # Cancel the existing timer
+    Process.cancel_timer(state.timer_ref)
+
+    timer_ref = schedule_flush()
+    {:noreply, %{state | timer_ref: timer_ref}}
+  end
+
+  defp schedule_flush do
+    Process.send_after(self(), :flush_ets, 7_500)
```

Comment:
The default prometheus scrape interval mentioned [here](https://prometheus.io/docs/prometheus/latest/configuration/configuration/) is one minute so it could be more efficient to have a flushing interval higher than that. On the other hand, one minute is long and an application generating a lot of metrics could accumulate a lot in this timing interval.

Reply:
Most deployments that I see are usually scraping every 5 or every 10 seconds....so I picked something in the middle. And for the reason you mentioned I didn't want to wait the whole minute to accumulate metrics.

Reply:
Just to be sure I understand correctly: with a scraping every 10 seconds, the ets will be flushed at least once between two scrapes right?

Reply:
That's correct. It'll look something like so:
```
      |-------7.5s--------|----2.5s----|
Scrape 1            Cron Flush      Scrape 2
```

In addition, every time an actual scrape occurs from Prometheus, the existing GenServer timer is canceled and a new one is created via the `handle_cast`. Another option would be to add a configuration value so that you can tweak this behavior, but I am hesitant to keep adding more config values.